### PR TITLE
TUG OF WAR USING BACKTRACKING IN C++  ISSUE: #3685

### PR DESCRIPTION
**Pull request by jaimehta19**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(YES I Have read all the guidelines )

### Description

(This problem is very important for competitive programming point of view .I have read all the guidelines and worked according to that.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

**Comment by ananddasani** [comment]
> ### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
> (YES I Have read all the guidelines )
> 
> ### Description
> (This problem is very important for competitive programming point of view .I have read all the guidelines and worked according to that.)
> 
> ### Checklist
> * [ YES ] I've read the contribution guidelines.
> * [ YES ] I've checked the issue list before deciding what to submit.
> * [ YES ] I've edited the `README.md` and link to my code.
> 
> ## Related Issues or Pull Requests
> (No issues.)

(Under checklist)
You have to type x , to check

**Comment by ananddasani** [comment]
Time complexity is not present in the code 

Write it at the end

**Comment by ananddasani** [comment]
you have to write complexities at the end of file

**Comment by ananddasani** [comment]
> ### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
> (YES I Have read all the guidelines )
> 
> ### Description
> (This problem is very important for competitive programming point of view .I have read all the guidelines and worked according to that.)
> 
> ### Checklist
> * [ YES ] I've read the contribution guidelines.
> * [ YES ] I've checked the issue list before deciding what to submit.
> * [ YES ] I've edited the `README.md` and link to my code.
> 
> ## Related Issues or Pull Requests
> (No issues.)

like this 
 [x] I've read the contribution guidelines.
 [x] I've checked the issue list before deciding what to submit.
 [x] I've edited the `README.md` and link to my code.

**Comment by jaimehta19** [comment]
@ananddasani done sir 👍

File: C-Plus-Plus/cp/Tug_of_water.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Tug_of_water.cpp:8]: Add a beginner description for the Code

File: C-Plus-Plus/cp/Tug_of_water.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+using namespace std;
+
+void traverse(int* arr, int i, int n, int no_of_selected, int sum, int& diff, bool* take, vector<int>& result, vector<int> current, int current_sum)
+{
+    //if the current position is greater than or equals to n
+    if (i >= n)
+        return;
+
+    //if the value of difference is greater than
+    //Sum/2-current sum of the elements of Subset no.1
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Tug_of_water.cpp]: Use multi-line comments

File: C-Plus-Plus/cp/Tug_of_water.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        }
+    }
+    cout << "}" << endl;
+}
+
+int main()
+{
+    int t;
+
+    cout << "Test case : ";
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Tug_of_water.cpp]: Test Case?

File: C-Plus-Plus/cp/Tug_of_water.cpp [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    Output: set1: {1, 4} set2: {2, 3}
+
+    
+	N=10
+    1 4 8 6 -7 -10 87 54 16 100
+    
+	Output: set1: {{ 4, -7, -10, 87, 54 } set2: {1, 8, 6, 16, 100 }
+   
+    */
+    // TIME COMPLEXITY OF TUG OF WAR IS O(N^2) and SPACE COMPLEXITY IS O(1)
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Tug_of_water.cpp]: Format and indent it properly. Keep time and space complexity within the previous comment block

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/Tug_of_water.cpp (hunk 1 above)
Comment (jaimehta19) [reply C-Plus-Plus/cp/Tug_of_water.cpp:8]: @HarshCasper done sir

File: C-Plus-Plus/cp/Tug_of_water.cpp (hunk 4 above)
Comment (jaimehta19) [reply C-Plus-Plus/cp/Tug_of_water.cpp]: @HarshCasper  done sir

File: C-Plus-Plus/cp/Tug_of_water.cpp (hunk 2 above)
Comment (jaimehta19) [reply C-Plus-Plus/cp/Tug_of_water.cpp]: @HarshCasper done sir

**Review by jaimehta19 — COMMENTED** [review]
@HarshCasper sir all the recommended changes by you have been done .Please review my PR

File: C-Plus-Plus/cp/Tug_of_water.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	N=4
+    1 2 3 4
+    Output: set1: {1, 4} set2: {2, 3}
+    
+	N=10
+    1 4 8 6 -7 -10 87 54 16 100
+    
+	Output: set1: {{ 4, -7, -10, 87, 54 } set2: {1, 8, 6, 16, 100 }
+   
+    */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Tug_of_water.cpp]: Left align and format it properly. Also remove extra lines of code

File: C-Plus-Plus/cp/Tug_of_water.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/Tug_of_water.cpp:107]: Remove this extra line of code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**